controller_manager:
  ros__parameters:
    update_rate: 10  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

joint_pd_controller:
  ros__parameters:
    type: fr3_custom_controllers/GravityCompensationController
    joints:
      - fr3_joint_linear
      - fr3_joint1
      - fr3_joint2
      - fr3_joint3
      - fr3_joint4
      - fr3_joint5
      - fr3_joint6
      - fr3_joint7
    gains:
      fr3_joint_linear: {p: 3000.0, d: 100.0}
      fr3_joint1:       {p: 3000.0, d: 100.0}
      fr3_joint2:       {p: 3000.0, d: 100.0}
      fr3_joint3:       {p: 3000.0, d: 100.0}
      fr3_joint4:       {p: 3000.0, d: 100.0}
      fr3_joint5:       {p: 3000.0, d: 100.0}
      fr3_joint6:       {p: 3000.0, d: 100.0}
      fr3_joint7:       {p: 3000.0, d: 100.0}

    command_interfaces:
      - effort

    state_interfaces:
      - position
      - velocity